src/lib.rs

```diff
@@ -625,6 +628,45 @@ pub trait Itertools : Iterator {
         Coalesce::new(self, eq)
     }
 
+    /// Remove non-unique elements from the iterator.
```

Comment:
I think it's better to formulate it not as removing, but maybe filtering. Something about producing an iterator of unique elements (for the whole sequence).